src/main/scala/org/broadinstitute/dsde/workbench/avram/api/AvramRoutes.scala

```diff
@@ -21,33 +22,63 @@ class AvramRoutes {
 
   private val log = Logger.getLogger(getClass.getName)
 
-  private val database = Avram.database
-
   @ApiMethod(name = "ping", httpMethod = "get", path = "ping")
   def ping: Pong = {
     Pong()
   }
 
+  @ApiMethod(name = "authPing", httpMethod = "get", path = "authPing")
+  def authPing(r: HttpServletRequest): Pong = {
+    val transformed = for {
+      userInfo <- extractUserInfo(r)
+    } yield {
+      log.info(userInfo.userEmail)
+      log.info(userInfo.userSubjectId)
+      Pong()
+    }
+
+    unsafeRun(transformed)
+  }
+
   // TODO: remove this endpoint when we have more meaningful ways to test database queries
   @ApiMethod(name = "now", httpMethod = "get", path = "now")
   def now: Now = {
-    // Explicitly use a Future to make sure the implicit ExecutionContext is being used
-    Now(Await.result(Future(fetchTimestampFromDBWithSlick()), Duration.Inf))
+    val transformed = for {
+      now <- runQuery(sql"select now()".as[String])
+    } yield Now(now.head)
+
+    unsafeRun(transformed)
   }
 
   // TODO: move/merge this endpoint into a status API
   @ApiMethod(name = "dbPoolStats", httpMethod = "get", path = "dbPoolStats")
   def dbPoolStats: DbPoolStats = {
-    val dataSource = Avram.dbcpDataSource
-    val result = for {
-      totalConnections <- database.run(
-        sql"select count(*) from pg_stat_activity where pid <> pg_backend_pid() and usename = current_user".as[Int])
-    } yield DbPoolStats(dataSource.getNumActive, dataSource.getNumIdle, totalConnections.head)
-    Await.result(result, Duration.Inf)
+    val transformed = for {
```

Comment:
This illustrates the general pattern for an api method:
```
def api: Foo = {
  val transformed = for {
    a <- someAvramResult
    b <- anotherAvramResult
  } yield Foo(a, b)

  unsafeRun(transformed)
}
```

Reply:
`unsafeRun` doesn't need any other parameters because they'd be bundled up in `transformed`, right? I'm trying to keep in mind the notion that we're first building the program to run (represented by `transformed`) and then running it.

Reply:
Correct, `transformed` is a _description_ of the program in terms of the effect stack, and `unsafeRun` actually runs it.